src/qilib/configuration_helper/adapters/d5a_instrument_adapter.py

```diff
@@ -46,11 +49,34 @@ def __init__(self, address: str, instrument_name: Optional[str] = None) -> None:
             raise SpanValueError('D5a instrument has span unequal to "4v bi"')
 
     def apply(self, config: PythonJsonStructure) -> None:
-        """ Step values for dacs should be part of configuration."""
-        super().apply(config)
+        """ Does not apply config, except for instrument name, to device, but compares config to device settings.
+
+        Args:
+            config: Containing the instrument configuration.
+
+        Raises:
+            ConfigurationError: If config does not match device configuration .
+
+        """
         unit = config['dac1']['unit']
         self._instrument.set_dac_unit(unit)
         dac_parameters = {param: values for param, values in config.items() if param[0:3] == 'dac'}
         for dac, values in dac_parameters.items():
             self._instrument[dac].step = values['step']
             self._instrument[dac].inter_delay = values['inter_delay']
+
+        device_config = self.read(True)
+
+        for parameter in config:
+            if parameter in self._instrument.parameters and hasattr(self._instrument.parameters[parameter], 'set'):
+                if 'value' in config[parameter]:
+                    self._assert_value_matches(config[parameter]['value'], device_config[parameter]['value'], parameter)
+
+        if 'name' in config:
+            self._instrument.name = config['name']
```

Comment:
You cannot update the name of a `qcodes.Instrument` by setting it, this will not update the name in the station (and probably other places).  I even think it would be better if the Instrument name was read-only.

Reply:
If the sequence of operation is load & apply config, then create the station. And thereafter reloading the hardware_kernel picks up the latest changes. If, so then use case would pick the latest names.

Reply:
@sinhaneelam Not sure what you are saying here, let's discuss offline. Renaming a qcodes instrument will not work.

Reply:
@peendebak  sure